src/AltinnCore/Runtime/RestControllers/InstancesController.cs

```diff
@@ -0,0 +1,210 @@
+using System;
+using System.IO;
+using System.Net.Http;
+using System.Text;
+using System.Threading.Tasks;
+using Altinn.Platform.Storage.Models;
+using AltinnCore.Common.Clients;
+using AltinnCore.Common.Configuration;
+using AltinnCore.Common.Services.Interfaces;
+using AltinnCore.Runtime.ModelBinding;
+using AltinnCore.Runtime.RestControllers;
+using Microsoft.AspNetCore.Authorization;
+using Microsoft.AspNetCore.Http;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.Extensions.Logging;
+using Microsoft.Extensions.Options;
+using Newtonsoft.Json;
+
+namespace AltinnCore.Runtime
+{
+    /// <summary>
+    /// Handles and dispatches operations to platform storage for the application instance resources
+    /// </summary>
+    [Route("{org}/{app}/instances")]
+    [ApiController]
+    public class InstancesController : ControllerBase
+    {
+        private readonly IAuthorization authorization;
+        private readonly ILogger<InstancesController> logger;
+        private readonly PlatformSettings platformSettings;
+        private readonly HttpClient storageClient;
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="InstancesController"/> class
+        /// </summary>
+        public InstancesController(
+            IAuthorization authorizationService,
+            ILogger<InstancesController> logger,
+            IOptions<PlatformSettings> platformSettings,
+            IHttpClientAccessor httpClientAccessor)
+        {
+            this.authorization = authorizationService;
+            this.logger = logger;
+            this.platformSettings = platformSettings.Value;
+            this.storageClient = httpClientAccessor.StorageClient;
+        }
+
+        /// <summary>
+        ///  Gets one application instance from platform storage.
+        /// </summary>
+        /// <param name="instanceOwnerId">the instance owner id (partyId)</param>
+        /// <param name="instanceGuid">the instance guid</param>
+        /// <returns></returns>
+        [HttpGet("{instanceOwnerId:int}/{instanceGuid:guid}")]
+        [Authorize]
+        [Produces("application/json")]
+        public async Task<ActionResult> Get(int instanceOwnerId, Guid instanceGuid)
```

Comment:
Where is this message called from?
I guess you had a discussion about how to differentiate between app and studio.. how did you conclude to solve that for this controller?

Reply:
This message is called from other client systems. Hence it is not expected to be called from studio.